gemelli/tests/test_preprocessing.py

```diff
@@ -2,120 +2,129 @@
 import numpy as np
 import pandas as pd
 import numpy.testing as npt
-from deicode.preprocessing import rclr
-from gemelli.preprocessing import build
 from skbio.stats.composition import clr
+from deicode.preprocessing import rclr as matrix_rclr
+from gemelli.preprocessing import build, rclr, rclr_matrix
 
 
 class Testpreprocessing(unittest.TestCase):
 
     def setUp(self):
-
-        self.cdata1 = np.array([[2, 2, 6],
-                                [4, 4, 2]])
-
-        self.cdata2 = [[3, 3, 0], [0, 4, 2]]
-
-        self.true2 = np.array([[0.0,    0.0,        np.nan],
-                               [np.nan, 0.34657359, -0.34657359]])
-
-        self.T_true2 = [[[0.,          0.,          0.],
-                         [0.,          0.,          0.]],
-                        [[-0.14384104, -0.14384104, -0.14384104],
-                         [0.14384104,   0.14384104,  0.14384104]],
-                        [[0.,          0.,          0.],
-                         [0.,          0.,          0.]]]
-
-        self.bad1 = np.array([1, 2, -1])
-
-        self.tenres = np.array([[[1,  2,  3],
-                                 [4,  5,  6]],
-                                [[7,  8,  9],
-                                 [10, 11, 12]],
-                                [[13, 14, 15],
-                                 [16, 17, 18]]])
-
-        self.clrres = np.array([[[-1.76959918],
-                                 [-1.076452],
-                                 [-0.67098689],
-                                 [0.17631097],
-                                 [0.30984236],
-                                 [0.4276254],
-                                 [0.79535018],
-                                 [0.86945815],
-                                 [0.93845102]],
-                                [[-0.88804481],
-                                 [-0.66490126],
-                                 [-0.48257971],
-                                 [0.02824592],
-                                 [0.1235561],
-                                 [0.21056747],
-                                 [0.49824955],
-                                 [0.55887417],
-                                 [0.61603258]]])
-        self.t = build()
-        self._rclr = rclr()
+        # test dense
+        self.count_data_one = np.array([[2, 2, 6],
+                                        [4, 4, 2]])
+        # test with zeros
+        self.count_data_two = np.array([[3, 3, 0],
+                                        [0, 4, 2]])
+        # test dense tensor
+        self.tensor_true = np.array([[[1, 2, 3],
+                                      [4, 5, 6]],
+                                     [[7, 8, 9],
+                                      [10, 11, 12]],
+                                     [[13, 14, 15],
+                                      [16, 17, 18]]])
         pass
 
     def test_build(self):
```

Comment:
This is testing a lot of things. Its easier to isolate issues if you split up test cases more. 
Maybe something like `test_build` and `test_build_errors`. Could probably be split up further.